components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/util/AssertionBuildingTest.java

```diff
@@ -0,0 +1,266 @@
+
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.security.UnrecoverableKeyException;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.Matchers.anyBoolean;
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.*;
+
+
+@PowerMockIgnore({"javax.net.*"})
+@PrepareForTest({IdentityUtil.class, IdentityTenantUtil.class, IdentityProviderManager.class,
+        SSOServiceProviderConfigManager.class})
+
+public class AssertionBuildingTest extends PowerMockTestCase {
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new PowerMockObjectFactory();
+    }
+
+    @Mock
+    private RealmService realmService;
+
+    @Mock
+    private TenantManager tenantManager;
+
+    @Mock
+    private IdentityProviderManager identityProviderManager;
+
+    @Mock
+    private IdentityProvider identityProvider;
+
+    @Mock
+    private KeyStoreManager keyStoreManager;
+
+    @Mock
+    private FederatedAuthenticatorConfig federatedAuthenticatorConfig;
+
+    @Mock
+    SSOServiceProviderConfigManager ssoServiceProviderConfigManager;
+
+    @Mock
+    private X509Credential x509Credential;
+
+    private final String CLAIM_URI1 = "http://wso2.org/claimuri1";
+    private final String CLAIM_URI2 = "http://wso2.org/claimuri2";
+    private final String CLAIM_VALUE1 = "ClaimValue1";
+    private final String CLAIM_VALUE2 = "ClaimValue2";
+
+    @Test
+    public void testBuildAssertion() throws Exception {
+
+        boolean claim1Found = false;
+        boolean claim2Found = false;
+        Assertion assertion = buildAssertion();
+        assertEquals(assertion.getSubject().getNameID().getValue(), TestConstants.TEST_USER_NAME, "Correct subject is" +
+                " not set to assertion");
+        assertEquals(assertion.getAttributeStatements().size(), 1, "Expected to have one attribute statement");
+        assertEquals(assertion.getAttributeStatements().get(0).getAttributes().size(), 2, "Expected to have two " +
+                "attributes");
+        for (int i = 0; i < 2; i++) {
+            if (CLAIM_URI1.equalsIgnoreCase(assertion.getAttributeStatements().get(0).getAttributes().get(i).getName
+                    ())) {
+                claim1Found = true;
+            }
+            if (CLAIM_URI2.equalsIgnoreCase(assertion.getAttributeStatements().get(0).getAttributes().get(i).getName
+                    ())) {
+                claim2Found = true;
+            }
+        }
+        assertTrue(claim1Found, "Claim1 was not found in the assertion");
+        assertTrue(claim2Found, "Claim2 was not found in the assertion");
```

Comment:
Why can't we do assertEquals here? It will reduce the code lines considerably!
And, you do not need a for loop to just iterate twice, do you?

Reply:
If we are to do assertEquals we need to check both attribute statements. We cannot guarantee the order of the attribute statements. That's why I iterate through all attributes and see whether the expected attributes are there.

Reply:
How about the following? I'm simply trying to reduce the complexity. Your logic is correct.

Map map = new HashMap();
map.put(assertion.getAttributeStatements().get(1).getAttributes().get(0).getName());
map.put(assertion.getAttributeStatements().get(1).getAttributes().get(1).getName());

assertTrue(map.contains(CLAIM_URI1));
assertTrue(map.contains(CLAIM_URI2));